consensus/checks.go

```diff
@@ -13,38 +15,55 @@ import (
 	"github.com/pkg/errors"
 )
 
+const MessageSigCacheSize = 1024
+
 // MaxBlockNumDiff limits the received block number to only 100 further from the current block number
 const MaxBlockNumDiff = 100
 
+var (
+	// BLSPubKeyCache is the Cache of the Deserialized BLS PubKey
+	MessageSigCache, _ = lru.New(MessageSigCacheSize)
+)
+
 // verifyMessageSig verify the signature of the message are valid from the signer's public key.
-func verifyMessageSig(signerPubKey *libbls.PublicKey, message *msg_pb.Message) error {
+func verifyMessageSig(senderKey *bls.SerializedPublicKey, message *msg_pb.Message) error {
 	signature := message.Signature
 	message.Signature = nil
 	messageBytes, err := protobuf.Marshal(message)
 	if err != nil {
 		return err
 	}
 
+	// Look up cache first
+	MessageSigKey := append(senderKey[:], messageBytes[:]...)
```

Comment:
Message could contain the block message that can be several megabytes. Saving the content string as key may require memory of several gigabytes, which is not resource friendly. Suggest to use the hash of the message instead of the content itself.